plugins/in_dummy/in_dummy.c

```diff
@@ -103,14 +248,23 @@ static int in_dummy_collect(struct flb_input_instance *ins,
         return -1;
     }
 
-    if (ctx->fixed_timestamp == FLB_FALSE) {
-        msgpack_sbuffer_init(&mp_sbuf);
-        gen_msg(ins, in_context, &mp_sbuf);
-        flb_input_log_append(ins, NULL, 0, mp_sbuf.data, mp_sbuf.size);
-        msgpack_sbuffer_destroy(&mp_sbuf);
+    if (strcasecmp(ctx->event_type, "log") == 0) {
+        ins->event_type = FLB_INPUT_LOGS;
+
+        if (ctx->fixed_timestamp == FLB_FALSE) {
+            msgpack_sbuffer_init(&mp_sbuf);
+
+            gen_msg(ins, in_context, &mp_sbuf);
+            flb_input_log_append(ins, NULL, 0, mp_sbuf.data, mp_sbuf.size);
+
+            msgpack_sbuffer_destroy(&mp_sbuf);
+        }
+        else {
+            flb_input_log_append(ins, NULL, 0, ctx->data.log.mp_sbuf.data, ctx->data.log.mp_sbuf.size);
+        }
     }
-    else {
-        flb_input_log_append(ins, NULL, 0, ctx->mp_sbuf.data, ctx->mp_sbuf.size);
+    else if (strcasecmp(ctx->event_type, "trace") == 0) {
```

Comment:
It is better to call strcasecmp only on init function and save as an enum value.